### PR TITLE
Issue #1980 Use `testify/assert` assertions in some remaining unit tests

### LINKED ISSUE #1980: Some unit tests do not use testify
### General information
We still do not use `testify/assert` for assertions in some of the units tests. Maybe we missed it in the review. Refer the old issue for using testify https://github.com/minishift/minishift/issues/250 and the patch at https://github.com/minishift/minishift/commit/838f4d50ce96e022f4990d37ba56704869b2f250

  * Minishift version: Latest Master i.e.  acb558a890b0e3afb93304e57a418b355cbbda63 

### Actual
```
$ git grep 'Fatal'
cmd/minishift/cmd/delete_test.go:               t.Fatalf("Expected cache dir '%s' to be deleted", state.InstanceDirs.Cache)
cmd/minishift/cmd/delete_test.go:               t.Fatalf("Expected cache dir '%s' to still exist", state.InstanceDirs.Cache)
cmd/testing/cli/harness.go:             t.Fatal(err)
cmd/testing/cli/harness.go:             t.Fatalf("Unexpected error during setup: %s", err.Error())
cmd/testing/cli/harness.go:                     t.Fatalf("Expected error '%s'. Got '%s'.", expectedErrorMessage, tee.StderrBuffer.String())
cmd/testing/cli/harness.go:                     t.Fatalf("Expected exit code %d. Got %d.", expectedExitCode, code)
cmd/testing/cli/harness.go:             t.Fatal("The called method unexpectedly called atexit.")
cmd/testing/cli/harness.go:             t.Fatal(err)
cmd/testing/cli/harness.go:             t.Fatal(err)
cmd/testing/cli/harness.go:             t.Fatal(err)
cmd/testing/cli/viper.go:                       t.Fatalf("Config %s not read correctly: %v", testOption.ConfigValue, err)
pkg/minikube/kubeconfig/config_test.go:         t.Fatalf("Error delete file %s", testSystemConfigPath)
pkg/minikube/machine/client_test.go:            log.Fatal(err)
pkg/minishift/addon/manager/addon_manager_test.go:              t.Fatal(fmt.Sprintf("No error expected, but got: \"%s\"", err.Error()))
pkg/minishift/provisioner/minishift_detector_test.go:           t.Fatalf("Error starting ssh server: %s", err)
pkg/minishift/registration/redhat_test.go:              t.Fatalf("Error starting ssh server: %s", err)
pkg/testing/assert.go:          t.Fatalf("String slices do not match. Expected %v, got %v", expected, actual)
pkg/testing/assert.go:          t.Fatalf("String slices differ in length. Expected %v, got %v", expected, actual)
pkg/testing/assert.go:                  t.Fatalf("%s and %s differ. Expected %v, got %v", expected[i], actual[i], expected, actual)
pkg/testing/runner_mock.go:     t.Fatalf("Expected: %+v, Got: %+v\n", argStringOption, expectCommands[0].commandArgs)
pkg/testing/tee_test.go:                t.Fatalf("Unexpected error: %s", err.Error())
pkg/util/utils_test.go:         t.Fatal("Unpexpected OS")
pkg/util/utils_test.go:         t.Fatal("Unpexpected OS")
pkg/util/utils_test.go:         t.Fatal("Unpexpected OS")
```

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #1980

### CODE REVIEW

File: pkg/minishift/addon/manager/addon_manager_test.go [hunk 1]
```diff
@@ -114,7 +115,8 @@ func TestAddVarDefaultsToContext(t *testing.T) {
 
 	testAddonMap := getTestAddonMap("test", "test description", expectedVarName, varDefault, "")
 
-	addOnMeta := getAddOnMetadata(testAddonMap, t)
+	addOnMeta, err := addon.NewAddOnMeta(testAddonMap)
+	assert.NoError(t, err, "Failed to create new addon meta")
```
Comment (praveenkumar) [inline pkg/minishift/addon/manager/addon_manager_test.go]: "Failed to create new addon meta" => "Failed to create addon meta" no need to put `new` here `NewAddonMeta` is just a function which returns add-on metadata.

Comment (anjannath) [reply pkg/minishift/addon/manager/addon_manager_test.go]: I thought it was some kind of a constructor, so went with `new`, anyways will change it.

File: pkg/minishift/addon/manager/addon_manager_test.go [hunk 2]
```diff
@@ -144,10 +147,11 @@ func TestVerifyMissingRequiredVariablesInContext(t *testing.T) {
 
 	testAddonMap := getTestAddonMap("test", "test description", expectedVarName, "", "")
 
-	addOnMeta := getAddOnMetadata(testAddonMap, t)
+	addOnMeta, err := addon.NewAddOnMeta(testAddonMap)
+	assert.NoError(t, err, "Failed to create addon meta")
```
Comment (praveenkumar) [inline pkg/minishift/addon/manager/addon_manager_test.go]: 👍

**Comment by anjannath** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
Thanks @anjannath

**Pull request merged**